### PR TITLE
new kallisto module

### PR DESCRIPTION
**Pull request by darogan**
Hopefully its in the clusterflow way of doing things

Tested and works on a SLURM HPC

### CODE REVIEW

**Comment by ewels** [comment]
Brilliant, thanks! I'll have a quick look through the code now..

File: modules/kallisto.cfmod.pl [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		# kallisto alignment typically takes less than 10 hours per BAM file
+		# This is probably inaccurate? May need tweaking.
+		return CF::Helpers::minutes_to_timestamp ($num_files * 14 * 60);
+	}
+);
+
+# Help text
+my $helptext = "".("-"x17)."\n Kallisto\n".("-"x17)."\n
+kallisto is a program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment. On benchmarks with standard RNA-Seq data, kallisto can quantify 30 million human reads in less than 3 minutes on a Mac desktop computer using only the read sequences and a transcriptome index that itself takes less than 10 minutes to build. Pseudoalignment of reads preserves the key information needed for quantification, and kallisto is therefore not only fast, but also as accurate as existing quantification tools. In fact, because the pseudoalignment procedure is robust to errors in the reads, in many benchmarks kallisto significantly outperforms existing tools.\n\n";
```
Comment (ewels) [inline modules/kallisto.cfmod.pl]: Impressively long line :) I usually put line breaks into these..

File: modules/kallisto.cfmod.pl [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+   die "\n\n###CF Error: No kallisto transcriptome index found in run file $cf{run_fn} for job $cf{job_id}. Exiting.. ###";
+} else {
+    warn "\nPseudoaligning against $cf{refs}{kallisto}\n\n";
+}
+
+open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";
+
+# Print version information about the module.
+warn "---------- Kallisto version information ----------\n";
+warn `kallisto | grep "^kallisto"`;
```
Comment (ewels) [inline modules/kallisto.cfmod.pl]: Can just do `kallisto version` ?

Comment (ewels) [inline modules/kallisto.cfmod.pl]:
```
~ » module load kallisto
~ » kallisto version
kallisto, version 0.42.4
```

File: modules/kallisto.cfmod.pl [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+		}
+
+                my $output_dir = $file."_kallisto_output";
+		my $output_fn  = $file."_kallisto.bam";
+                
+                my $command = "kallisto quant -t $cf{cores} --pseudobam --single -i $cf{refs}{kallisto} -o $output_dir -b 100 $file | samtools view -Sb - > $output_fn";
+		warn "\n###CFCMD $command\n\n";
+
+		if(!system ($command)){
+			# Bowtie worked - print out resulting filenames
```
Comment (ewels) [inline modules/kallisto.cfmod.pl]: Missed a find and replace spot ;)

File: modules/kallisto.cfmod.pl [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+			}
+
+                        my $output_dir = $files[0]."_kallisto_output";
+                        my $output_fn  = $files[0]."_kallisto.bam";
+
+                        my $command = "kallisto quant -t $cf{cores} --pseudobam -i $cf{refs}{kallisto} -o $output_dir -b 100 $files[0] $files[1] | samtools view -Sb - > $output_fn";
+			warn "\n###CFCMD $command\n\n";
+
+			if(!system ($command)){
+				# Bowtie worked - print out resulting filenames
```
Comment (ewels) [inline modules/kallisto.cfmod.pl]: ^^

**Comment by ewels** [comment]
A couple of very minor things but otherwise looks perfect, thanks for this!

**Comment by darogan** [comment]
You are welcome, glad I can contribute

**Comment by ewels** [comment]
Ok great! If you agree with my comments above then it'd be great if you could push a new commit addressing them, otherwise let me know and I'll merge.

**Comment by darogan** [comment]
I made the changes, and pushed a new commit. Let me know if you need any other changes

**Comment by ewels** [comment]
Looks great, thanks!

**Pull request merged**